bridge/engine/EngineStreamDirector.h

```diff
@@ -475,7 +475,10 @@ class EngineStreamDirector
 
         const auto pinMapItr = _pinMap.find(toEndpointIdHash);
         const bool fromPinnedEndpoint = pinMapItr != _pinMap.end() && isSsrcFromParticipant(pinMapItr->second, ssrc);
-        const auto assignedQuality = (fromPinnedEndpoint ? viewer->pinQualityLevel : viewer->unpinQualityLevel);
+
+        // In case slides are used - limit video resolution to save CPU on recipients.
+        const auto assignedQuality =
+            (fromPinnedEndpoint && 0 == _slidesBitrateKbps) ? viewer->pinQualityLevel : viewer->unpinQualityLevel;
```

Comment:
Interesting way of "detecting" slides presence. We do have the contentType in the adding of participant streams. But I guess this is more dynamically checking that slides are actually sent also, not just announced.